### ISSUE TITLE
Release Build Failure, "Task :notifee_react-native:bundleReleaseAar FAILED"

### ISSUE THREAD

**Issue by joebarrero** [notifee/react-native-notifee#151]
I'm using React Native version 0.63.2 with Notifee 0.12.3

When using classpath('com.android.tools.build:gradle:4.0.1') in the build.gradle file, ./gradlew assembleRelease fails with the following error and message:

```
> Task :notifee_react-native:bundleReleaseAar FAILED
w: Detected multiple Kotlin daemon sessions at build/kotlin/sessions

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':notifee_react-native:bundleReleaseAar'.
> Direct local .aar file dependencies are not supported when building an AAR. The resulting AAR would be broken because the classes and Android resources from any local .aar file dependencies would not be packaged in the resulting AAR. Previous versions of the Android Gradle Plugin produce broken AARs in this case too (despite not throwing this error). The following direct local .aar file dependencies of the :notifee_react-native project caused this error: 

```
Changng the to build.grade file to, classpath('com.android.tools.build:gradle:3.6.4'), seemed to fix the issue.

I'm not sure if this is a bug, my lack of experience, or a combination of both but I thought it might be important enough to post my experience here.

The error seems to imply that creating a release bundle with the earlier version Grade Tools can lead to a broken bundle.  Is there something that I should do to assemble the release build the latest version of the Gradle Tools?

**Comment by mikehardy** [comment]
I am using the latest notifee release, gradle 4.0.1, as well as multiple other modules that use kotlin (react-native-apple-authentication, react-native-webview) without issue, so this is unexpected

If you could post a reproduction on github based off a clean project from `react-native init` that could help

You may want to define `kotlinVersion = 1.4.10` in your android/build.gradle ext block https://github.com/facebook/react-native/blob/83777cb4fb5dda89c430b7eff9cd1f28d2577831/template/android/build.gradle#L4-L10 - I have that and can vouch it works at least

**Comment by joebarrero** [comment]
@mikehardy 

Thank you for your response.  I tried again using Gradle 4.0.1 and updating Kotlin to 1.4.10.  I received the same error.  I created a clean and barebones instance of the project that can be reviewed at https://github.com/joebarrero/notifee-test and this version also produced the same error.

I added all of the relevant information regarding my build in the Readme.md file.  I appreciate your help.

**Comment by mikehardy** [comment]
How unfortunate! I'll give it a shot and see what I can see. Sorry you've run in to this trouble

**Comment by mikehardy** [comment]
:thinking: 

Does not reproduce:

```
mike@isabela:~/work/Invertase/customers/notifee-test (master *) % npm run android

> notifeetest@0.0.1 android /home/mike/work/Invertase/customers/notifee-test
> react-native run-android

info Running jetifier to migrate libraries to AndroidX. You can disable it using "--no-jetifier" flag.
Jetifier found 974 file(s) to forward-jetify. Using 12 workers...
info JS server already running.
info Installing the app...
Starting a Gradle Daemon (subsequent builds will be faster)

> Configure project :notifee_react-native
:notifee_react-native package.json found at /home/mike/work/Invertase/customers/notifee-test/node_modules/@notifee/react-native/package.json
:notifee_react-native:version set from package.json: 0.12.3 (0,12,3 - 12003)
:notifee_react-native:android.compileSdk using custom value: 29
:notifee_react-native:android.targetSdk using custom value: 29
:notifee_react-native:android.minSdk using custom value: 16
:notifee_react-native:reactNativeAndroidDir /home/mike/work/Invertase/customers/notifee-test/node_modules/react-native/android

> Task :app:compileDebugJavaWithJavac

> Task :app:installDebug
Installing APK 'app-debug.apk' on 'TestingAVD_API-29_TAG-PLAY(AVD) - 10' for app:debug
Installed on 1 device.

BUILD SUCCESSFUL in 26s
41 actionable tasks: 41 executed
info Connecting to the development server...
8081
info Starting the app on "emulator-5554"...
Starting: Intent { cmp=com.notifeetest/.MainActivity }
mike@isabela:~/work/Invertase/customers/notifee-test (master *) % git remote -v
origin  https://github.com/joebarrero/notifee-test.git (fetch)
origin  https://github.com/joebarrero/notifee-test.git (push)
```

How are you launching it? Via Android Studio or the command line?

**Comment by joebarrero** [comment]
@mikehardy 

It runs just fine.  It's when I try to create the APK that it fails.

```
cd android
./gradlew assembleRelease
```

**Comment by mikehardy** [comment]
My apologies - you noted that in the first line (or so) of your report. I was moving to quickly while attempting to diagnose. I have reproduced the problem! Which is bad (there's a problem...) but also great because now I see it.

**Comment by mikehardy** [comment]
I knew this reminded me of something!

Use `./gradlew :app:assembleRelease`

https://github.com/notifee/react-native-notifee/issues/68#issuecomment-682117223

Can you confirm this works @joebarrero ?

**Comment by joebarrero** [comment]
That worked! Thank you.

**Comment by mikehardy** [comment]
I am still baffled why `:app:<target>` is needed vs `<target>` but we have no other reported problems from folks using this. It feels like it should be called out in a document somewhere, I'll open a separate issue for that

Glad you're moving forward!

**Comment by fachri-adakerja** [comment]
Hi @mikehardy,

thanks for your work on this issue, my company uses the Notifee on our apps and 1 of our company app is currently broken because of this issue since we just upgrade to target API Lever 30.

We updating the API Level to 30 because of an announcement from google, below

```
Starting August 2021, new apps will be required to target API level 30 (Android 11) and use the Android App Bundle publishing format. Starting November 2021, all app updates will be required to target API level 30 (Android 11)
```

We hope this issue can be fixed and the Notifee team is able to do the release for a fix before November. So my company is still able to release a new version of our apps.

Thanks & Best regards @mikehardy


Below is the log

```bash
* What went wrong:
Execution failed for task ':notifee_react-native:bundleReleaseAar'.
> Direct local .aar file dependencies are not supported when building an AAR. The resulting AAR would be broken because the classes and Android resources from any local .aar file dependencies would not be packaged in the resulting AAR. Previous versions of the Android Gradle Plugin produce broken AARs in this case too (despite not throwing this error). The following direct local .aar file dependencies of the :notifee_react-native project caused this error: /Users/fachri/Projects/AdaKerja/steve-mobile-js/node_modules/@notifee/react-native/android/libs/notifee_core_release.aar

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':notifee_react-native:bundleReleaseAar'.
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:38)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:372)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:359)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:352)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:338)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
```

**Comment by helenaford** [comment]
Fixed in release v2.0.0 (https://notifee.app/react-native/docs/release-notes#200)

**Comment by mikehardy** [comment]
Should be noted this became mandatory with the move to android targetSdkVersion 30, not sure why? But this should keep everyone moving along as they move those targets up
